server/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -230,21 +255,41 @@ private static Map<String, Set<URL>> readSecuredFiles(
         return Collections.unmodifiableMap(securedFiles);
     }
 
-    private static void addSpeciallySecuredFile(Map<String, Set<URL>> securedFiles, String path) {
-        Set<URL> attemptedToGrant = securedFiles.put(path, Set.of());
-        if (attemptedToGrant != null) {
-            throw new IllegalStateException(attemptedToGrant + " tried to grant access to special config file " + path);
+    private static void readSecuredPermissions(
+        Environment environment,
+        URL url,
+        Permission p,
+        Map<String, Set<URL>> securedFiles,
+        Map<String, Set<URL>> securedSettingKeys
+    ) {
+        String securedFileName = extractSecuredName(p, SecuredFileAccessPermission.class);
+        if (securedFileName != null) {
+            Path securedFile = environment.configFile().resolve(securedFileName);
```

Comment:
Same here, let's make sure this is actually rooted in the config directory

Reply:
I've added a subpath check